### ISSUE TITLE
Implement typing test

### ISSUE THREAD

**Issue by ytturi** [GeeksCAT/anem-per-feina#52]
**Is your feature request related to a problem? Please describe.**
We are lacking a typing test

**Describe the solution you'd like**
Implement a test using mypy

**Describe alternatives you've considered**
A good approach will be that the test will pass as long as the newly modified files pass mypy.
We can achieve this by selecting the files that have been modified on a later date than when we implement this issue.
Git always keeps the file creation/modify date, so we can make use of it.

We may want to implement a small (bash) script to collect the files that we want to test or adapt this test (at least for now).

**Additional context**
Consider using `annotations` and `typing` in all files. Also consider using the `TYPE_CHECKING` from `typing` to import types only used in annotations:

```python
from __future__ import annotations
from typing import List, TYPE_CHECKING

if TYPE_CHECKING:
  from this.project import CustomType

def check_this_method(foo: List[CustomType]) -> int:
    return len(foo)
```

Mypy documentation: https://mypy.readthedocs.io/en/stable/
Basically:
- Mypy will check all files (checking the imports as well) to see if we are using type annotations
- Mypy will check the typing for all the objects, parameters, return values...

**Comment by ytturi** [comment]
I'll try to get this done so we ensure that the code is typed from the start

**Comment by ytturi** [comment]
Also I will use a different approach as the creation date is not kept correctly depending on the file system. I think it's not reliable.
I will use a commit hash instead, so all files modified after that hash should pass the mypy checks.
